src/ssmDocument/commands/createDocumentFromTemplate.ts

```diff
@@ -66,23 +68,26 @@ export async function createSsmDocumentFromTemplate(): Promise<void> {
 
     const selection = picker.verifySinglePickerOutput(choices)
 
-    // User pressed escape and didn't select a template
-    if (selection === undefined) {
-        return
-    }
-
     try {
-        logger.debug(`User selected the ${selection.label} template.`)
-        const textDocument: vscode.TextDocument = await openTextDocumentFromSelection(selection)
-        vscode.window.showTextDocument(textDocument)
+        // User pressed escape and didn't select a template
+        if (selection === undefined) {
+            result = 'Cancelled'
+        } else {
+            logger.debug(`User selected the ${selection.label} template.`)
```

Comment:
In general the variable part of a log message should go at the end, where possible. Also don't need punctuation (log messages are not prose).
```suggestion
            logger.debug(`User selected template: ${selection.label}`)
```